recirq/qml_lfe/dynamics_fuzz_test.py

```diff
@@ -0,0 +1,30 @@
+import os
+import pytest
+import tempfile
+import numpy as np
+
+
+def test_shadows_dont_seperate(tmpdir):
+    pyfile = os.path.join(os.path.dirname(__file__), "learn_dynamics_C.py")
+    os.system(
+        f"python3 {pyfile} --n=6 --depth=5 --n_data=20 --batch_size=20 --save_dir={tmpdir}"
+    )
```

Comment:
Never use os.system. Use the subprocess module.

Also this seems like a really weird way to test the script. Why not call the `main()` method? there's no guarantee that `python3` will refer to the current interpreter running the test and could result in some real head-scratchers

Reply:
All this test is depending on is that the file is spit out after being run. Calling main is tricky with absl because it uses `app.run` to inject the values of the flags into your program right before it gets run. It was tricky enough to get working in tests that I figured this was fine.

Reply:
Usually this is a sign that things should be split out further.  If you make main into flag resolution and then make a new function _execute_circuits(depth, save_dir, n_data, batch_size, ...) then you could get rid of the subprocess call, which seems like it will cause us problems in testing in the future if CI processes change.  This could be done in a later PR if necessary.